extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/Job.java

```diff
@@ -94,9 +94,12 @@ public static void scheduleEvent(String thingUID, AstroThingHandler astroHandler
         if (thingNull || astroHandlerNull || eventAtNull || eventsNull || channelIdNull || events.isEmpty()) {
             return;
         }
-        AstroChannelConfig config = astroHandler.getThing().getChannel(channelId).getConfiguration()
-                .as(AstroChannelConfig.class);
-        Calendar instant = applyConfig(eventAt, config);
+        Calendar instant = eventAt;
+        if (!configAlreadyApplied) {
+            AstroChannelConfig config = astroHandler.getThing().getChannel(channelId).getConfiguration()
+                    .as(AstroChannelConfig.class);
+            instant = applyConfig(eventAt, config);
+        }
```

Comment:
Any reason to initialize instant twice if `configAlreadyApplied == false`?

What about:
```Java
final Calendar instant;
if (!configAlreadyApplied) {
    AstroChannelConfig config = astroHandler.getThing().getChannel(channelId).getConfiguration()
            .as(AstroChannelConfig.class);
    instant = applyConfig(eventAt, config);
} else {
    instant = eventAt;
}
```